pkg/lib/sidecars/pods/filter_test.go

```diff
@@ -0,0 +1,63 @@
+package pods_test
+
+import (
+	"context"
+	"github.com/kyma-project/istio/operator/pkg/lib/sidecars/pods"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("Evaluate restart", func() {
+	It("should should return false when pod has custom image annotation", func() {
+		pod := createPodWithProxySidecar("test-pod", "test-namespace", "1.21.0", map[string]string{"sidecar.istio.io/proxyImage": "istio/proxyv2:1.22.0"})
```

Comment:
For the test data to be more consistent, the custom annotation should use the same version as the proxy sidecar. That is 1.21.0 in this test.
```suggestion
		pod := createPodWithProxySidecar("test-pod", "test-namespace", "1.21.0", map[string]string{"sidecar.istio.io/proxyImage": "istio/proxyv2:1.21.0"})
```

Reply:
thanks for catching that